acceptancetests/assess_persistent_storage.py

```diff
@@ -1,139 +1,428 @@
 #!/usr/bin/env python
-"""Testing persistent storage function of Juju."""
+"""Testing Juju's persistent storage feature."""
 
 from __future__ import print_function
 
-import argparse
-import yaml
-import logging
-from subprocess import CalledProcessError
+import os
 import sys
+import time
+import json
+import logging
+import argparse
+
+from jujucharm import local_charm_path
 
 from deploy_stack import (
     BootstrapManager,
-    get_random_string,
+    # TODO: for data verification:
+    # get_random_string,
     )
+
 from utility import (
     JujuAssertionError,
     add_basic_testing_arguments,
     configure_logging,
     )
-from jujucharm import local_charm_path
 
 __metaclass__ = type
-
-
 log = logging.getLogger("assess_persistent_storage")
 
 
-def assess_persistent_storage(client):
-    ensure_storage_remains_after_application_removal()
-
-
-def ensure_storage_remains_after_application_removal(client):
-    """Storage created during a deploy must persist after application removal.
-
-    Steps taken to test:
-      - Deploy the dummy storage charm
-      - Set config and ensure data is stored on storage.
-      - Remove application, taking note of the remaining storage name
-      - Re-deploy new charm using persisted storage
-      - Ensure data has remained.
-
-    :param client: ModelClient object to deploy the charm on.
+def get_storage_list(client):
+    raw_output = client.get_juju_output(
+        'storage', '--format', 'json', include_e=False)
+    # it's not a bug for juju storage --format json output to be empty
+    # if there is no storage exists.
+    if raw_output == '':
+        storage_list=storage_output=''
+    else:
+        storage_output = json.loads(raw_output)
```

Comment:
In this case I prefer to use if else because the if condition do 2 things: check the raw_output to decide load it or not, and set storage_list, storage_output to empty string if raw_output is empty.
Empty raw_output is not an error here, which wouldn't be suitable for raising exception.